Update libcantera-devel

ischoegl opened the pull request:
This PR includes some minor updates:

- Include Fortran modules in `libcantera-devel`
- ~Create "clean" sample folders for `cxx` and `f90`~
- Install `man` pages in `cantera`
- Ensure that `libboost` is a dependency for `libcantera-devel`
- Fix extra level in `matlab` folder structure
- Add SCons option `package_build = True`

Closes #37.

bryanwweber commented:
> * Ensure that `libboost` is a dependency for `libcantera-devel`

This shouldn't be needed, as we only rely on the header portions of the library? The concern I have is that this will then bring in `libboost` for the `cantera` package too (because `cantera` depends on `libcantera-devel`).

ischoegl commented:
> > * Ensure that `libboost` is a dependency for `libcantera-devel`
> 
> This shouldn't be needed, as we only rely on the header portions of the library? The concern I have is that this will then bring in `libboost` for the `cantera` package too (because `cantera` depends on `libcantera-devel`).

Since `libcantera-devel` depends on `libcantera`, there shouldn’t be a concern? I can confirm once the packages are done building. (This is still WIP after all 😆).

bryanwweber commented:
Ah, if `cantera` depends on `libcantera`, then `libcantera-devel` is separate. I don't think I realized we'd made that split. All should be fine then!

ischoegl commented:
@bryanwweber ... it is ready now. I also just confirmed that 
```
conda install -c file:///opt/conda/envs/conda-bld/conda-bld/ libcantera-devel
```
and 
```
conda install -c file:///opt/conda/envs/conda-bld/conda-bld/ cantera
```
have the expected dependencies.

bryanwweber commented on cantera/build_devel.bat:
```diff
@@ -4,6 +4,13 @@ ECHO ************************
+RD /S /Q "%PREFIX%/share/cantera/samples"
+
+ROBOCOPY "%SRC_DIR%/samples/cxx" "%PREFIX%/share/cantera/samples/cxx" /S /E
+DEL /F "%PREFIX%/share/cantera/samples/cxx/SCons*"
+DEL /F "%PREFIX%/share/cantera/samples/cxx/*Make*"
```
These seem useful, what do we need to do to be able to include them?

ischoegl replied:
Not sure. `conda-build` isn't able to patch them properly, and leaving the `*.in` will confuse people. I think it's best to point users to the C++ guide and leave them out (at least for the time being). Adding a comment.

bryanwweber commented on cantera/build_py.bat:
```diff
@@ -20,6 +20,7 @@ IF ERRORLEVEL 1 EXIT 1
 IF ERRORLEVEL 1 EXIT 1
 
 ROBOCOPY "%SRC_DIR%\samples\python" "%PREFIX%\share\cantera\samples\python" /S /E
+ROBOCOPY "%SRC_DIR%\platform\posix\man" "%PREFIX%\share\man\man1" /S /E
```
Does Windows understand man files?

ischoegl replied:
Probably not. Although Cantera installs them :wink:

Deleting.

bryanwweber reviewed (changes requested):
Two small things ☺️ I wish I had more time to be productive with Cantera!

ischoegl commented:
@bryanwweber ... thanks! Things are taken care of. I triggered a GH Action to build the packages just to make sure.

ischoegl commented:
Hm. Need to take care of `cantera.mod` for Fortran.

bryanwweber commented on cantera/build_devel.bat:
```diff
@@ -4,6 +4,20 @@ ECHO ************************
+REM Instead, just copy *.cpp sources
+RD /S /Q "%PREFIX%/share/cantera/samples"
+ROBOCOPY "%SRC_DIR%/samples/cxx" "%PREFIX%/share/cantera/samples/cxx" /S /E
+DEL /F "%PREFIX%/share/cantera/samples/cxx/SConscipt"
```
```suggestion
DEL /F "%PREFIX%/share/cantera/samples/cxx/SConscript"
```

bryanwweber commented on cantera/build_devel.bat:
```diff
@@ -4,6 +4,20 @@ ECHO ************************
+ROBOCOPY "%SRC_DIR%/samples/cxx" "%PREFIX%/share/cantera/samples/cxx" /S /E
+DEL /F "%PREFIX%/share/cantera/samples/cxx/SConscipt"
+DEL /F "%PREFIX%/share/cantera/samples/cxx/SConstruct.in"
+DEL /F "%PREFIX%/share/cantera/samples/cxx/Makefile.in"
```
This is probably an upstream issue, but the `.in` files shouldn't be installed at all, SCons should be filling in the templates.

ischoegl replied:
In this case, I am copying from `%SRC_DIR%`, which do have the templates. Hopefully, with #37 fixed, all of this will go away ...

bryanwweber replied:
Ah, I missed that. I don't think you should copy from the SRC_DIR, it should be from the installed location, right? And it shouldn't require any copying, if everything is installed to the right place.

ischoegl replied:
This is on purpose, as the build process creates makefiles in individual sub folders, which are difficult to scrub. I don’t see what speaks against copying from the source in this case as a temporary fix for #37?

ischoegl replied:
Honestly, the only alternative I see here is to not remove the files at all. The files contents are dubious, but we do consider them being ‘experimental’.

bryanwweber commented on cantera/meta.yaml:
```diff
@@ -13,6 +13,8 @@ requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+    # At the moment, the GH toolchain does not provide Fortran for Windows
```
GH, meaning GitHub? These compilers are provided by Conda... I guess except on Windows where it's just a shim for Visual Studio. So I think the bigger issue is that there isn't a libre Fortran compiler for Windows at all...

ischoegl replied:
:+1: Removed "GH"

bryanwweber reviewed (changes requested):
Another couple small clarifications here 😊

ischoegl commented:
@bryanwweber ... fixed things per comments. I also caught one extra-level introduced by copying in some folders of the `cantera-matlab` package on *nix when copying, which is also fixed.

ischoegl commented:
@bryanwweber ... I'd suggest to table the discussion on whether or not to package 'broken' `Makefile`'s. The issue is now documented - see #37 - and without it, this PR still introduces some features that are worthwhile (see top).

ischoegl commented:
@bryanwweber …. As things stands, this PR is independent of #37 and Cantera/cantera#1366. As such, this is converged on my side.

ischoegl commented:
Using artifacts from a CI run on [GitHub Actions](https://github.com/Cantera/conda-recipes/actions/runs/2923612813), I have confirmed that #37 is resolved on Linux. Tests on windows revealed issues beyond #37, although some of the build routes are broken even for a regular install (see Cantera/cantera#1375 etc.) and thus need to be taken care of upstream first. I haven't had a chance to test on macOS yet, but don't expect any surprises as the recipe is largely identical to that of Linux.
